integration/mediation-tests/tests-transport/src/test/java/org/wso2/carbon/esb/passthru/transport/test/PreserveLocationHeaderTestCase.java

```diff
@@ -0,0 +1,122 @@
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.automation.engine.context.AutomationContext;
+import org.wso2.carbon.automation.engine.context.TestUserMode;
+import org.wso2.carbon.automation.test.utils.http.client.HttpRequestUtil;
+import org.wso2.carbon.utils.ServerConstants;
+import org.wso2.esb.integration.common.utils.ESBIntegrationTest;
+import org.wso2.esb.integration.common.utils.common.ServerConfigurationManager;
+import org.wso2.esb.integration.common.utils.servers.WireMonitorServer;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * Test case for preserve Http Location header when the following property is configured in passthru-http.properties
+ * http.headers.preserve=Location
+ *
+ * https://wso2.org/jira/browse/ESBJAVA-3677
+ */
+public class PreserveLocationHeaderTestCase extends ESBIntegrationTest {
+
+    private ServerConfigurationManager serverConfigurationManager;
+    private WireMonitorServer wireMonitorServer;
+
+    @BeforeClass(alwaysRun = true)
+    public void init() throws Exception {
+        super.init();
+        verifyProxyServiceExistence("PreserveLocationHeaderTestProxy");
+        serverConfigurationManager = new ServerConfigurationManager(
+                new AutomationContext("ESB", TestUserMode.SUPER_TENANT_ADMIN));
+        applyPassThroughProperty();
+        wireMonitorServer = new WireMonitorServer(6771);
+        wireMonitorServer.start();
+    }
+
+    /**
+     * Test for preserve Location header with/without the http.headers.preserve=Location property
+     * @throws Exception
+     */
+    @Test(groups = "wso2.esb", description = "Test to verify preserve 'Location' header")
+    public void testPreserveLocationHeader() throws Exception {
+
+        String proxyServiceUrl = getProxyServiceURLHttp("PreserveLocationHeaderTestProxy");
+
+        String requestPayload = "<?xml version='1.0' encoding='UTF-8'?>\n" +
+                "<soapenv:Envelope xmlns:soapenv=\"http://schemas.xmlsoap.org/soap/envelope/\">\n" +
+                "   <soapenv:Header/>\n" +
+                "   <soapenv:Body/>\n" +
+                "</soapenv:Envelope>";
+
+        Map<String, String> headers = new HashMap<String, String>();
+        headers.put(HttpHeaders.CONTENT_TYPE, "text/xml");
+        headers.put(HTTPConstants.HEADER_SOAP_ACTION, "urn:mediate");
+        headers.put("Location", "http://localhost:8000");
+
+        HttpRequestUtil.doPost(new URL(proxyServiceUrl), requestPayload, headers);
+
+        String wireResponse = wireMonitorServer.getCapturedMessage();
+        String[] wireResponseList = wireResponse.split(System.lineSeparator());
+
+        Assert.assertTrue(wireResponse.contains("Location"));
+        for (String line : wireResponseList) {
```

Comment:
use a boolean variable to flag to check whether the location is not changed. then assert the variable